transaction.go

```diff
@@ -0,0 +1,71 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+)
+
+// A Transaction is used to push data from one matrix server to another matrix
+// server.
+type Transaction struct {
+	// The ID of the transaction.
+	TransactionID TransactionID `json:"transaction_id"`
+	// The server that sent the transaction.
+	Origin ServerName `json:"origin"`
+	// The server that should receive the transaction.
+	Destination ServerName `json:"destination"`
+	// The millisecond posix timestamp on the origin server when the transction
+	// was created.
+	OriginServerTS Timestamp `json:"origin_server_ts"`
+	// The IDs of the most recent transactions sent by the server.
```

Comment:
> sent by the server

Ambiguous. Do you mean "sent by the Destination server" or "sent by the Origin server"? It's not immediately clear as both could be valid (if they are the most recent txns sent by us then it forms a DAG, if they are the most recent txns received then it serves as an ACK).